apps/ledger-live-desktop/tests/specs/speculos/receive.address.spec.ts

```diff
@@ -49,3 +49,70 @@ for (const account of accounts) {
     );
   });
 }
+
+test.describe("Receive", () => {
+  const account = Account.TRX_2;
+  test.use({
+    userdata: "speculos-tests-app",
+    speculosCurrency: account.currency,
+  });
+  test(
+    `${account.currency} empty balance Receive displays address activation warning message`,
+    {
+      annotation: {
+        type: "TMS",
+        description: "B2CQA-1551",
+      },
+    },
+    async ({ app }) => {
+      await addTmsLink(getDescription(test.info().annotations).split(", "));
+
+      await app.layout.goToAccounts();
+      await app.accounts.navigateToAccountByName(account.accountName);
+      await app.account.expectAccountVisibility(account.accountName);
+
+      await app.account.clickReceive();
+      await app.modal.continue();
+      await app.receive.verifyWarningMessageContent(
+        "You first need to send at least 0.1 TRX to this address to activate it. Learn more",
+      );
+    },
+  );
+});
+
+const accountsWithToken: Map<Account, string> = new Map([
+  [Account.TRX_1, "TRC10/TRC20"],
+  [Account.ETH_1, "Ethereum"],
+  [Account.BSC_1, "BEP20"],
+]);
+
+accountsWithToken.forEach((message, account) => {
+  test.describe("Receive", () => {
+    test.use({
+      userdata: "speculos-tests-app",
+      speculosCurrency: account.currency,
+    });
+
+    test(
+      `[${account.currency.name}] Receive`,
+      {
+        annotation: {
+          type: "TMS",
+          description: "B2CQA-249, B2CQA-651",
+        },
+      },
+      async ({ app }) => {
+        await addTmsLink(getDescription(test.info().annotations).split(", "));
+
+        await app.layout.goToAccounts();
+        await app.accounts.navigateToAccountByName(account.accountName);
+        await app.account.expectAccountVisibility(account.accountName);
+        await app.account.clickReceive();
+        await app.receive.verifyWarningMessageContent(
```

Comment:
would it make sense to add this check to the main receive test case to avoid launching electron + speculos just for warning message verification?

Reply:
done